sql/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java

```diff
@@ -267,6 +273,55 @@ public void testSnapshotWithMetadataDoesNotDeleteExistingStuff() throws Exceptio
             "foo| 1.2\n"));
     }
 
+    @Test
+    public void testSnapshotWithMetadataConcurrentlyModified() throws Exception {
+        int shards = randomFrom(1, 3, 5);
+        int replicas = randomIntBetween(2, 10);
+        long documents = randomLongBetween(2, 100);
+
+        execute("CREATE TABLE test (" +
+                "  id long primary key)" +
+                "clustered into " + shards + " shards with (column_policy = 'dynamic', number_of_replicas=" + replicas +
+                ")");
+
+        ensureYellow();
+
+        ActionFuture<SQLResponse> createSnapshot = null;
+        // Insert data with dynamic column creation so we trigger a dynamic mapping update for each of them
+        for (var i = 0; i < documents; i++) {
+            execute("INSERT INTO test (id, field_" + i + ") VALUES (?, ?)", new Object[][]{{i, "value_" + i},});
+            if (createSnapshot == null) {
```

Comment:
I think a table `refresh` is needed here, otherwise the test could be flaky. (ES does it by changing the refresh policy on the bulk request which we can't)

Reply:
Ok, sounds reasonable. thanks for pointing out.